### ISSUE TITLE
HeadlessError: Error executing command to extract phantom ports

### ISSUE THREAD

**Issue by DenisIzmaylov** [baudehlo/node-phantom-simple#76]
Hello everyone!
Thanks for the project. Like it. But when I try to execute `node-phantom-simple` inside of Docker container (dokku DigitalOcean-droplet) it causes error:
```bash
[HeadlessError: Error executing command to extract phantom ports: Error: Command failed: /bin/sh -c if which ss > /dev/null; then ss -nlp | grep "[,=]41,"; else netstat -nlp | grep "[[:space:]]41/"; fi
```
It looks as a same issue like at https://github.com/Shippable/support/issues/884

**Comment by puzrin** [comment]
Seems you don't have `ss` and `netstat`. Could you instal one of those? Or may be `lsof` is available?

**Comment by DenisIzmaylov** [comment]
That's right:
```bash
# netstat
bash: netstat: command not found
```
The same issue for `lsof`. It's basic Dokku droplet in DigitalOcean. I'm tryin' to solve it. But there is question - should we have an external dependency to `netstat` tool and other?

**Comment by puzrin** [comment]
You need to install package with `ss` or `netstat` into your docklet (`iproute` or `net-tools`). I don't know anything about DO droplets, and can't advice details. But you can see that this project executes tests in travis-ci docker containers.

**Comment by DenisIzmaylov** [comment]
I've just solved this issue and install `netstat` inside of the container. But still the same errors occurred. I tried to go inside of the container and execute the command directly (through `dokku run app-name bash`):
```bash
root@a0257e81ee59:~# /bin/sh -c if which ss > /dev/null; then ss -nlp | grep "[,=]44,"; else netstat -nlp | grep "[[:space:]]44/"; fi
bash: syntax error near unexpected token `then'
```
But when I try run the same command through `dokku run app-name sh`:
```bash
# if which ss > /dev/null; then ss -nlp | grep "[,=]44,"; else netstat -nlp | grep "[[:space:]]44/"; fi
```
It outputs nothing.

**Comment by DenisIzmaylov** [comment]
I tried to run node.js and execute command directly by `exec()`. It produces error:
```
{ [Error: Command failed: /bin/sh -c if which ss > /dev/null; then ss -nlp | grep "[,=]44,"; else netstat -nlp | grep "[[:space:]]44/"; fi
]
  killed: false,
  code: 1,
  signal: null,
  cmd: '/bin/sh -c if which ss > /dev/null; then ss -nlp | grep "[,=]44,"; else netstat -nlp | grep "[[:space:]]44/"; fi' }
```

**Comment by svvac** [comment]
Hi, I just struggled against the same  issue. Thought I'd share in case it helps someone.

In my case, it was `ss -nlp` that subrepticely segfaulted (while `ss` by itself did not). Fixed it by removing the conditional  in the `switch` case to always use `netstat` (but that's obviously a quick hack).

Btw, the reason why you get the Parse error when you paste the command in your shell is that the command separatorts (`;`) are interpreted by your shell and not the subshedd (i.e. `/bin/sh`). Wrap the complete command between quotes and it should do the trick.

    # /bin/sh -c 'if which ss > /dev/null; then ss -nlp | grep "[,=]44,"; else netstat -nlp | grep "[[:space:]]44/"; fi'

Also note that you need to have a process with PID 44 for the above command to output anything.

Cheers.

**Comment by puzrin** [comment]
@svvac https://github.com/baudehlo/node-phantom-simple/blob/master/node-phantom-simple.js#L187 could PR fix for this line?

**Comment by svvac** [comment]
I managed to make it work by using the following command:

    cmd = 'ss -nlp | grep "[,=]%d," || netstat -nlp | grep "[[:space:]]%d/"';

It may not be an optimal solution when only netstat is available, but it allows to fallback when `ss` exists but fails.

**Comment by puzrin** [comment]
@svvac thanks for your help!

Released 2.2.1 with your fix.
